packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterUpdateUserAttributeRequest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_auth_cognito.types
+
+import androidx.annotation.NonNull
+import com.amazonaws.amplify.amplify_core.exception.ExceptionMessages
+import com.amazonaws.amplify.amplify_core.exception.InvalidRequestException
+import com.amplifyframework.auth.AuthUserAttribute
+
+data class FlutterUpdateUserAttributeRequest(val map: HashMap<String, *>) {
+
+    val attribute: AuthUserAttribute = formatUpdateUserAttribute(map["attribute"] as HashMap<String, *>);
+
+    private fun formatUpdateUserAttribute(@NonNull rawAttribute: HashMap<String, *>): AuthUserAttribute {
+        val value = rawAttribute["value"].toString();
+        val key: String = rawAttribute["userAttributeKey"] as String;
+        val attribute: AuthUserAttribute = createAuthUserAttribute(key, value);
+        return attribute;
+    }
+
+    companion object {
+        private const val validationErrorMessage: String = "UpdateUserAttributeRequest Request malformed."
+        fun validate(req : HashMap<String, *>?) {
+            if (req == null) {
+                throw InvalidRequestException(validationErrorMessage, ExceptionMessages.missingAttribute.format("request map"))
+            } else if (!req.containsKey("attribute") || req["attribute"] !is HashMap<*, *>) {
+                throw InvalidRequestException(validationErrorMessage, ExceptionMessages.missingAttribute.format( "attribute" ))
+            } else {
+                val attribute = req["attribute"] as HashMap<*, *>;
+                if (!attribute.containsKey("value")) {
+                    throw InvalidRequestException(validationErrorMessage, ExceptionMessages.missingAttribute.format( "value" ))
+                } else if (!attribute.containsKey("userAttributeKey")) {
```

Comment:
Does it also need to validate the `attribute["userAttributeKey"] !is String` like it does for value?

Reply:
In general we don't check the type of each attribute on the request (the exception being for anything we expect to be a HashMap, to avoid trying access a key/value on an object that isn't a HashMap). I think you could argue that we should though. I _think_ the reason that we generally don't check the type of everything is that the purpose of the validate methods is to fail fast when a mistake is made is serializing the object, opposed to being used as a way to enforce correct user input (that should, and is, done on the dart side with proper types).

The reason I _did_ add a type check for `value` is because Cognito lets customers define user attributes as a String or an Int. However, the iOS and Android SDKs expect a string regardless of what the type was set to in Cognito. I wanted to be very explicit that `value` should always be a String, and we should not be serializing it to an Int in any scenario.

I added a comment to note that iOS/Android SDKs expect strings regardless of the configuration in Cognito to at least make it clear why I have a type check on that specifically.

Reply:
cool, sounds good